src/libcharon/plugins/kernel_super/picomsg_conversions.c

```diff
@@ -65,6 +65,21 @@ void ts_to_prefix_string_for_logs(struct traffic_selector_t* ts,
 	pm_proto_prefix_to_string_for_logs(&route_prefix, str);
 }
 
+bool ts_to_prefix_string(struct traffic_selector_t* ts,
+						char str[MAX_IP_PREFIX_STR_LEN]) {
+	struct pm_proto_prefix prefix;
+	if(! ts_to_picomsg_prefix(ts, &prefix)) {
+		LOG(DBG_KNL, SL_LOG_ERR, "failed to get traffic selector");
+		return false;
+	}
+
+	if(! pm_proto_addr_to_string(&prefix.prefix, str)) {
```

Comment:
This only sets the address. You need the whole prefix (need a `pm_proto_prefix_to_string()`)